File: src/bmsx_cpp/core/system_machine.cpp
#include "system_machine.h"

namespace bmsx {
namespace {

const MachineManifest SYSTEM_MACHINE_MANIFEST = [] {
	MachineManifest manifest;
	manifest.namespaceName = "bmsx";
	manifest.viewportWidth = 256;
	manifest.viewportHeight = 212;
	manifest.canonicalization = CanonicalizationType::Lower;
	manifest.ufpsScaled = 50'000'000;
	manifest.cpuHz = 1'000'000;
	manifest.imgDecBytesPerSec = 26'214'400;
	manifest.dmaBytesPerSecIso = 8'388'608;
	manifest.dmaBytesPerSecBulk = 26'214'400;
	manifest.vdpRenderBudgetPerFrame = DEFAULT_VDP_RENDER_BUDGET_PER_FRAME;
	return manifest;
}();

} // namespace

const MachineManifest& defaultSystemMachineManifest() {
	return SYSTEM_MACHINE_MANIFEST;
}

const char* systemBootEntryPath() {
	return "res/bios/bootrom.lua";
}

} // namespace bmsx

File: src/bmsx/core/system_machine.ts
import type { MachineManifest } from '../rompack/rompack';

export const SYSTEM_ROM_NAME = 'bmsx-bios';
export const SYSTEM_BOOT_ENTRY_PATH = 'res/bios/bootrom.lua';

export const SYSTEM_MACHINE_MANIFEST: MachineManifest = {
	render_size: {
		width: 256,
		height: 212,
	},
	canonicalization: 'lower',
	namespace: 'bmsx',
	ufps: 50_000_000,
	specs: {
		cpu: {
			cpu_freq_hz: 1_000_000,
			imgdec_bytes_per_sec: 26_214_400,
		},
		dma: {
			dma_bytes_per_sec_iso: 8_388_608,
			dma_bytes_per_sec_bulk: 26_214_400,
		},
		vdp: {
			render_budget_per_frame: 512,
		},
	},
};
